File: animon/src/tileMain/Tile.java
package tileMain;
import java.awt.Graphics;
import java.awt.image.BufferedImage;

import main.Game;

import tiles.GrassTile;
import tiles.MeadowTile;
import tiles.RockTile;
import tiles.SandTile;
import tiles.TreeTile;

public abstract class Tile {
	protected BufferedImage image;
		public static Tile grass = new GrassTile();
		public static Tile sand = new SandTile();
		public static Tile rock = new RockTile();
		public static Tile meadow = new MeadowTile();
		public static Tile tree = new TreeTile();
		
	public Tile(String path) {
		image = Game.getLoader().load("/tiles/" + path + ".png");
	}

	public abstract void tick();

	public void render(Graphics g, int x, int y)
	{
			g.drawImage(image, x, y, Game.TILESIZE * Game.SCALE,
					Game.TILESIZE * Game.SCALE, null);	
	}
	
	public boolean isSolid()
	{
	return false;	
	}
	
	}

File: alnaria/Tile.cpp
#include "Game.hpp"

Tile::Tile(std::string path, bool solid)
{
	texture = new sf::Texture();
	texture->loadFromFile(path);
	this->solid = solid;
}

Tile::~Tile()
{
	delete texture;
}

Tile* Tile::colorToTile(sf::Color color)
{
	static Tile *grass = new Tile("png/tile/grass.png", false);
	static Tile *tree = new Tile("png/tile/tree.png", true);
	static Tile *sand = new Tile("png/tile/sandway.png", false);
	static Tile *water = new Tile("png/tile/water.png", false);
	static Tile *rock = new Tile("png/tile/rock.png", true);

	if (color == sf::Color(0, 255, 0))
		return grass;
	else if (color == sf::Color(20, 75, 20))
		return tree;
	else if (color == sf::Color(200, 200, 0))
		return sand;
	else if (color == sf::Color(0, 0, 255))
		return water;
	else if (color == sf::Color(75, 75, 75))
		return rock;

	return sand;
}

sf::Texture* Tile::getTexture()
{
	return texture;
}

bool Tile::isSolid()
{
	return solid;
}
